### DISCUSSION TITLE
Ideas - automatic close service, auto change id/password, linux taskbar, disable user input improvement

### DISCUSSION THREAD

**Discussion by owlysk** [rustdesk/rustdesk#3073] [Ideas]
Hi there. I Have some ideas about improvent your awesome app:

- _Add setting to automatic close service when main window closed!_
I understand that, some users want to have service always on, so they can connect everytime. But I think it will be nice to have option to stop service when main rundesk window closed (where is my ID,password, etc, like TeamViewer). Now I need to check everytime I use rustdesk if is service off or on before close program.

- _Automatic change id/password_
 Add setting to automatic change ID/password/both after program close (or computer is rebooted).

- _Gnome Dash to Panel (Linux, Ubuntu 22.04.1)_
it will be nice, if rustdesk has taskbar icon, which showed in Dash to Panel

- _Disable user input improvement_
I noticied that master can disable user input. I'ts good feature but I think, remote user must have some hotkey to disable this. Imagine that some scammer use rustdesk and disable user input. If I want to take control to my PC again I have tu turn off PC or plug off the power cable :(. If there will be some hotkey to take control again like CTRL+ALT+DEL it will be nice. Or when master activated disable user input feature, on client side shows dialog with hotkey to take control back for 5s.

**Comment by marbx** [comment]
## Regarding *Automatic change password*
I support  a  setting to automatically change password after program close, enabled by default.

Currently, a supported user must remember to change the password or else any supporters connected once can reconnect without password. This is too broad a trust given to a potentially increasing set of persons.
The supported user should make a conscious decision to leave the password unchanged after program close.
Also, TeamViewer and AnyDesk, are  changing the password after program close, so most users will have this expectation.

Changing the id should also be a settings option, but disabled by default.


Thank you

**Comment by SergeyMy** [comment]
1. password changes every time rustdesk is turned on
2. rustdesk does not need to start the service! use portable mode

Service is not running = remote access disabled!

**Comment by marbx** [comment]
I opened and closed RustDeck 1.1.9 on Debian and MacOS several times and always got the same password 
Which version are you using?

**Comment by SergeyMy** [comment]
1.1.9 is very very outdated! use:
https://github.com/rustdesk/rustdesk/releases/tag/nightly

**Comment by marbx** [comment]
Thank you for the link to the nightly builds.
Version 1.2.0 shows a "One time password", which stays the same after closing, same as in 1.1.9.
In other words it does no change every time RustDesk is turned on.

Will the password change after use?

**Reply by 21pages** [reply]
It will be regenerated after one connection is closed.

**Reply by marbx** [reply]
perfect, thank you

**Comment by owlysk** [comment]
I tried 1.2.0 AppImage version on Linux and onetime password wont to be generated. 

But this ability seems to be what I wanted :)
